dangerzone/isolation_provider/container.py

```diff
@@ -175,19 +190,42 @@ def install() -> bool:
                         p.stdin.write(chunk)
                 else:
                     break
-        p.communicate()
+        _, err = p.communicate()
+        if p.returncode < 0:
+            if err:
+                error = err.decode()
+            else:
+                error = "No output"
+            raise ImageInstallationException(
+                f"Could not install container image: {error}"
+            )
 
-        if not Container.is_container_installed():
-            log.error("Failed to install the container image")
+        if not Container.is_container_installed(raise_on_error=True):
             return False
 
         log.info("Container image installed")
         return True
 
     @staticmethod
-    def is_container_installed() -> bool:
+    def is_runtime_available() -> bool:
+        container_runtime = Container.get_runtime()
+        runtime_name = Container.get_runtime_name()
+        # Can we run `docker/podman image ls` without an error
+        with subprocess.Popen(
+            [container_runtime, "image", "ls"],
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.PIPE,
+            startupinfo=get_subprocess_startupinfo(),
+        ) as p:
+            _, stderr = p.communicate()
+            if p.returncode != 0:
+                raise NotAvailableContainerTechException(runtime_name, stderr.decode())
+            return True
+
+    @staticmethod
+    def is_container_installed(raise_on_error: bool = False) -> bool:
```

Comment:
Don't know about you, but I think this method is very confusing. It checks if the container is installed, but also somehow has a side-effect (removing the container image if the image ID does not match). I think we can do better here, but that would be out of scope of this PR.

Reply:
I agree: I was this 🤏🏼 close to propose a change there. Next time we touch here, I guess :+1: